databases/week2/library-database.sql

```diff
@@ -0,0 +1,53 @@
+use library;
+SET
+  NAMES utf8mb4;
+CREATE TABLE `author` (
+    `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `name` varchar(255) NOT NULL,
+    `email` varchar(255) NOT NULL
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `rack` (
+    `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `categary` varchar(255) NOT NULL
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `genre` (
+    `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `name` varchar(255) NOT NULL
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `member` (
+    `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `name` varchar(255) NOT NULL,
+    `phone` varchar(255) NULL,
+    `email` varchar(255) NOT NULL
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `book` (
+    `id` int(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY,
+    `name` varchar(255) NOT NULL,
+    `description` text NULL DEFAULT NULL,
+    `price` DATETIME NOT NULL,
+    `genre_id` int(10) unsigned NOT NULL,
+    `rack_id` int(10) unsigned NOT NULL,
+    CONSTRAINT `fk_genre` FOREIGN KEY (`genre_id`) REFERENCES `genre` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+    CONSTRAINT `fk_rack` FOREIGN KEY (`rack_id`) REFERENCES `rack` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `book_author` (
+    `book_id` int(10) unsigned NOT NULL,
+    `author_id` int(10) unsigned NOT NULL,
+    PRIMARY KEY(`book_id`, `author_id`),
+    CONSTRAINT `fk_book_book_author` FOREIGN KEY (`book_id`) REFERENCES `book` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+    CONSTRAINT `fk_author_book_author` FOREIGN KEY (`author_id`) REFERENCES `author` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+  ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+  CREATE TABLE `borrowLog` (
```

Comment:
this table is ok, but it could also be merged with the book_member_borrowLog table
basically you'd end up with a "borrow" table with the columns "id, book_id, member_id, borrow_date, return_date". if you do this you probably don't want to make (book_id, member_id) your primary key though, because then the same person wouldn't be able to borrow the same book twice.